demos/dev_mode_key_provisioning/include/aws_dev_mode_key_provisioning.h

```diff
@@ -23,87 +23,195 @@
-    uint8_t * pcClientPrivateKey;          /**< Pointer to the device private key in PEM format.  
-                                                See tools/certificate_configuration/PEMfileToCString.html 
-                                                for help with formatting.*/
-    uint32_t ulClientPrivateKeyLength;     /**< Length of the private key data, in bytes. */
-    uint8_t * pcClientCertificate;         /**< Pointer to the device certificate in PEM format.
-                                                See tools/certificate_configuration/PEMfileToCString.html 
-                                                for help with formatting.*/ 
-    uint32_t ulClientCertificateLength;    /**< Length of the certificate in bytes. */
+    uint8_t * pucClientPrivateKey;      /**< Pointer to the device private key in PEM format.
+                                         *   See tools/certificate_configuration/PEMfileToCString.html
+                                         *   for help with formatting.*/
+    uint32_t ulClientPrivateKeyLength;  /**< Length of the private key data, in bytes. */
+    uint8_t * pucClientCertificate;     /**< Pointer to the device certificate in PEM format.
+                                         *   See tools/certificate_configuration/PEMfileToCString.html
+                                         *   for help with formatting.*/
+    uint32_t ulClientCertificateLength; /**< Length of the device certificate in bytes. */
+    uint8_t * pucJITPCertificate;       /**< Pointer to the Just-In-Time Provisioning (JITP) certificate in 
+                                         *   PEM format.  If no JITP certificate exists,
+                                         *   this should be set to NULL.
+                                         *   See tools/certificate_configuration/PEMfileToCString.html
+                                         *   for help with formatting.
+                                         *   See https://tinyurl.com/y3qf3un5 for more information about 
+                                         *   getting started with J*/
+    uint32_t ulJITPCertifiateLength;    /**< Length of the Just-In-Time Provisioning (JITP) certificate in bytes.
+                                         *   If no JITP certificate exists, this should be set to 0. */
 } ProvisioningParams_t;
 
+
 /** \brief Provisions device with default credentials.
- * 
+ *
  * Imports the certificate and private key located in
  * aws_clientcredential_keys.h to device NVM.
  */
 void vDevModeKeyProvisioning( void );
 
-/** \brief Provisions device with provided credentials. 
- * 
+/** \brief Provisiong a device given a valid PKCS #11 session.
+*
+* \param[in] xSession       A valid PKCS #11 session.
+* \param[in] pxParams       Pointer to an initialized provisioning
+*                           structure.
+*
+* \return CKR_OK upon successful key creation.  PKCS #11 error code on failure.
+* Note that PKCS #11 error codes are positive.
```

Comment:
The last line of this comment is cruft, in my opinion.

Reply:
updated.